teuthology/orchestra/remote.py

```diff
@@ -210,25 +210,18 @@ def mktemp(self):
         """
         Make a remote temporary file
 
-        Returns: the name of the temp file created using
-                 tempfile.mkstemp
-        """
-        py_cmd = "import os; import tempfile; import sys;" + \
-            "(fd,fname) = tempfile.mkstemp();" + \
-            "os.close(fd);" + \
-            "sys.stdout.write(fname.rstrip());" + \
-            "sys.stdout.flush()"
-        args = [
-            'python',
-            '-c',
-            py_cmd,
-            ]
-        proc = self.run(
-            args=args,
-            stdout=StringIO(),
-            )
-        data = proc.stdout.getvalue()
-        return data
+        Returns: the path of the temp file created.
+        """
+        args = ['mktemp']
+
+        if suffix:
+            args.append('--suffix')
```

Comment:
I'm afraid you didn't declar suffix and parent arguments in the signature of this function.